Firestore/core/src/model/mutable_document.h

```diff
@@ -211,27 +200,25 @@ class MutableDocument {
   MutableDocument(DocumentKey key,
                   DocumentType document_type,
                   SnapshotVersion version,
-                  ObjectValue value,
+                  std::shared_ptr<const ObjectValue> value,
                   DocumentState document_state)
       : key_{std::move(key)},
         document_type_{document_type},
-        version_{std::move(version)},
+        version_{version},
         value_{std::move(value)},
         document_state_{document_state} {
   }
 
   DocumentKey key_;
   DocumentType document_type_ = DocumentType::kInvalid;
   SnapshotVersion version_;
-  ObjectValue value_;
+  std::shared_ptr<const ObjectValue> value_ = std::make_shared<ObjectValue>();
```

Comment:
Can you add a comment to explain the rationale of making this a shared pointer?

Reply:
Done